tests/asahirov/hw7_fibo.py

```diff
@@ -0,0 +1,33 @@
+"""
+Create decorator which would remember answers,
+and if answer was given, do not make a call to a function
+"""
+
+
+def remember_answer(func):
+    answer = {}
+
+    def remember(*args):
+        try:
+            hash(args)
+        except Exception:
+            raise TypeError("input args not hashable ")
+        if args in answer:
+            return answer[args]
+        else:
+            result = func(*args)
+            answer[args] = result
+            return result
+
+    return remember
```

Comment:
try to avoid nesting and multiple returns if possible

```
def remember_answer(func):
    answer = {}

    def remember(*args):
        try:
            hash(args)
        except Exception:
            raise TypeError("input args not hashable ")

        if args not in answer:
            result = func(*args)
            answer[args] = result
        return answer[args]

    return remember

```